pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1205,8 +1207,28 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 		if err != nil {
 			return err
 		}
-	}
 
+		if hasIstioSidecarInjectionEnabled(b.vmi) && proto == iptables.ProtocolIPv4 {
+			err = b.handler.NftablesAppendRule(proto, "nat", "output",
+				b.handler.GetNFTIPString(proto), "saddr", getEnvoyLoopbackAddress(),
```

Comment:
Same here; would you export this to a function ? `performDNATForEnvoyTrafficOnOutputChain` or something similar.

Everytime you rebase, I have to read this again, and re-learn what this does. I think encapsulating this in a function with a meaningful name would be helpful.

Also for the SNAT performed below.